src/clang-c-frontend/clang_c_convert.cpp

```diff
@@ -450,24 +450,31 @@ bool clang_c_convertert::get_var(const clang::VarDecl &vd, exprt &new_expr)
     return true;
 
   // Check if we annotated it to be have an infinity size
+  bool no_slice = false;
   if(vd.hasAttrs())
   {
     for(auto const &attr : vd.getAttrs())
     {
       if(const auto *a = llvm::dyn_cast<clang::AnnotateAttr>(attr))
       {
-        if(a->getAnnotation().str() == "__ESBMC_inf_size")
+        const std::string &name = a->getAnnotation().str();
+        if(name == "__ESBMC_inf_size")
         {
           assert(t.is_array());
           t.size(exprt("infinity", uint_type()));
         }
+        else if(name == "__ESBMC_no_slice")
+          no_slice = true;
       }
     }
   }
 
   std::string id, name;
   get_decl_name(vd, name, id);
 
+  if(no_slice)
```

Comment:
I prefer to add a flag to `symbolt` and set it here. That way we can see it when dumping the context/symbol table.

Reply:
I think that using it directly at config seems easier and more practical. Also, we can check if a symbol id is in the `no_slice` if we want to print it.